src/internal/components/autosuggest-input/index.tsx

```diff
@@ -297,7 +308,7 @@ const AutosuggestInput = React.forwardRef(
           loopFocus={loopFocus}
         >
           {open && dropdownContent ? (
-            <div ref={dropdownContentRef} className={styles['dropdown-content']}>
+            <div ref={dropdownContentRef} className={styles['dropdown-content']} onKeyDown={handleDropdownKeyDown}>
```

Comment:
There is an existing escape key handler already: https://github.com/cloudscape-design/components/blob/0ad2f7f8c55a6c48690b46ea0ae6a09bd1502f95/src/internal/components/autosuggest-input/index.tsx#L176-L179

Instead of building a duplicate handler, can you make the original handler work as expected

Reply:
good call, the handling here is a bit different for each one (like when pressing ESC within the recovery button it should focus back the input. If the ESC is pressed and you focused the input already, it should not focus it again).

I created like a shared function to handle these two cases and applied it.